### ISSUE TITLE
question: How is fluttium different from Maestro by mobile.dev?

### ISSUE THREAD

**Issue by bartekpacia** [wolfenrain/fluttium#10]
It looks very similar. I'm interested in seeing a feature comparison with [Maestro](https://maestro.mobile.dev).

**Comment by ttpho** [comment]
I have the same question.

**Comment by wolfenrain** [comment]
Maestro focusses on fully black box testing, testing applications for which u dont might not have access to the source code. On top of that Maestro is a UI automation framework, Fluttium on the other side is user flow testing, dedicated to your own application, making it more grey box testing.

And because it is tightly coupled to Flutter itself you also gain all it's benefits: 
- fully cross platform, while Maestro only has Android and iOS (for now)
- Live code changes while running your test flows

Sadly, currently at comes at the cost of not having native interactions which Maestro is capable of doing. But there are ways to solve that (as seen by [Patrol](https://patrol.leancode.co/))

You can already achieve everything Fluttium does by purely using the Flutter Integration Test framework, but Fluttium brings the focus back to user flows by applying the same kind of powerful declarative syntax that Maestro use.

Hope this helps explaining it a bit, if not just ask away!

TLDR: both are great for user flow testing but each uses different means for different goals.

**Comment by bartekpacia** [comment]
Thanks a lot for a detailed response @wolfenrain! This sounds interesting, I'll have to try it out. I think it would be great to create a page in the docs where a comparison between `fluttium` and other testing frameworks would be. We have something like this in [Patrol docs](https://patrol.leancode.co/comparison) (it's an early version of such page, but we'd like to make it better). But I understand that you've just released it and there are more urgent things to do.

Good luck with the project! :)

**Comment by wolfenrain** [comment]
I am indeed planning on doing that! Once the feature set settles down a bit I'll be writing a dedicated page for that, comparing frameworks but also give a more insight in when to use which one.

Thank you for your kind words btw, and if you have any feedback ya know where the issue reporting feature is ;-)

**Comment by lehmannk** [comment]
If then else decissions in the flow would be very much needed in order to run more complex tests. Anything in the queue for this?
